packages/block-library/src/playlist/edit.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * External Dependencies
+ */
+
+import { uniqBy } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	FormFileUpload,
+	IconButton,
+	Placeholder,
+	ServerSideRender,
+	Toolbar,
+	withNotices,
+	CheckboxControl,
+	PanelBody,
+	SelectControl,
+} from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	BlockControls,
+	InspectorControls,
+	mediaUpload,
+	MediaUpload,
+} from '@wordpress/editor';
+
+class PlaylistEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.initializePlaylist = this.initializePlaylist.bind( this );
+		this.uploadFromFiles = this.uploadFromFiles.bind( this );
+		this.onUploadFiles = this.onUploadFiles.bind( this );
+
+		// check for if ids is set to determine edit state
+		this.state = {
+			isEditing: ! this.props.attributes.ids,
+		};
+	}
+
+	initializePlaylist() {
+		window.wp.playlist.initialize();
```

Comment:
Noting that globals should be considered "evil" in this context. We want to eliminate any direct references to `window.wp` with an equivalent import if it's possible. I'm sensing that for playlists specifically, this may be difficult to accomplish. Since `block-library` is a [published module](https://www.npmjs.com/package/@wordpress/block-library), we should probably be conscious of any global dependencies we assume to exist by documenting them. Even in the context of WordPress, it's not obvious on the surface that we need playlist scripts to be available any time we load the `block-library` script.